apis/virtualkubelet/v1alpha1/shadowpod_types.go

```diff
@@ -24,15 +24,43 @@ type ShadowPodSpec struct {
 	Pod corev1.PodSpec `json:"pod,omitempty"`
 }
 
+// PodPhase describes the phase of the ShadowPod.
+type PodPhase string
+
+const (
+	// ShadowPodPending indicates a pending phase.
+	ShadowPodPending PodPhase = "Pending"
+	// ShadowPodRunning indicates running phase.
+	ShadowPodRunning PodPhase = "Running"
+	// ShadowPodSucceeded indicates succeeded phase.
+	ShadowPodSucceeded PodPhase = "Succeeded"
+	// ShadowPodFailed indicates failed phase.
+	ShadowPodFailed PodPhase = "Failed"
+	// ShadowPodUnknown indicates unknown phase.
+	ShadowPodUnknown PodPhase = "Unknown"
+)
+
+// ShadowPodStatus defines the observed state of ShadowPod.
+type ShadowPodStatus struct {
+	// Phase is the status of this ShadowPod.
+	// When the pod is created it is checked by the operator, which sets this field same as pod status.
+	// +kubebuilder:validation:Enum="Pending";"Running";"Succeeded";"Failed";"Unknown"
+	// +kubebuilder:default="Unknown"
+	Phase PodPhase `json:"phase"`
+}
+
 // +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
 // +genclient
 
 // ShadowPod is the Schema for the Shadowpods API.
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.phase`
```

Comment:
Is the `Age` automatically printed, or you need to add a `// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`` annotation?

Reply:
Done